src/quartz_api/internal/inputs/dataplatform/client.py

```diff
@@ -0,0 +1,294 @@
+            site_uuid,
+            dp.EnergySource.SOLAR,
+            authdata["sub"],
+        )
+        return generation
+
+    @override
+    async def post_site_generation(
+        self,
+        site_uuid: str,
+        generation: list[internal.ActualPower],
+        authdata: dict[str, str],
+    ) -> None:
+        raise NotImplementedError("Data Platform client doesn't yet support site writing.")
+
+    @override
+    async def save_api_call_to_db(self, url: str, authdata: dict[str, str]) -> None:
+        raise NotImplementedError("Data Platform client doesn't yet support API call logging.")
+
+    async def _get_actual_power_production_for_location(
+        self,
+        location: str,
+        energy_source: dp.EnergySource,
+        oauth_id: str,
+    ) -> list[internal.ActualPower]:
+        """Local function to retrieve actual values regardless of energy type."""
+        await self._check_user_access(
+            location,
+            energy_source,
+            dp.LocationType.SITE,
+            oauth_id,
+        )
+
+        start, end = get_window()
+        req = dp.GetObservationsAsTimeseriesRequest(
+            location_uuid=location,
+            observer_name="ruvnl",
+            energy_source=energy_source,
+            time_window=dp.TimeWindow(
+                start_timestamp_utc=start,
+                end_timestamp_utc=end,
+            ),
+        )
+        resp = await self.dp_client.get_observations_as_timeseries(req)
+        out: list[internal.ActualPower] = [
+            internal.ActualPower(
+                Time=value.timestamp_utc,
+                PowerKW=int(value.effective_capacity_watts * value.value_fraction / 1000.0),
+            )
+            for value in resp.values
+        ]
+
+        return out
+
+    async def _get_predicted_power_production_for_location(
+        self,
+        location: str,
+        energy_source: dp.EnergySource,
+        oauth_id: str,
+        forecast_horizon: ForecastHorizon = ForecastHorizon.latest,
+        forecast_horizon_minutes: int | None = None,
+        smooth_flag: bool = True,
+    ) -> list[internal.PredictedPower]:
+        """Local function to retrieve predicted values regardless of energy type."""
+        _ = await self._check_user_access(
+            location,
+            energy_source,
+            dp.LocationType.SITE,
+            oauth_id,
+        )
+
+        start, end = get_window()
+
+        if forecast_horizon == ForecastHorizon.latest or forecast_horizon_minutes is None:
+            forecast_horizon_minutes = 0
+        elif forecast_horizon == ForecastHorizon.day_ahead:
+            # The intra-day forecast caps out at 8 hours horizon, so anything greater than that is
+            # assumed to be day-ahead. It doesn't seem like it's as simple as just using 24 hours,
+            # from my asking around at least
+            forecast_horizon_minutes = 9 * 60
```

Comment:
Yea your right, its more complicated than this. I think we'll need a seperate sql query e.t.c. 

Do you mind add a github issue in here or the data-platform and adding a link to this

Reply:
Sure, link here - https://github.com/openclimatefix/data-platform/discussions/21